airflow/providers/amazon/aws/sensors/s3.py

```diff
@@ -131,6 +136,50 @@ def poke(self, context: Context):
         else:
             return all(self._check_key(key) for key in self.bucket_key)
 
+    def execute(self, context: Context) -> None:
+        """
+        Defers to Trigger class to poll for state of the job run until
```

Comment:
it defers to trigger only if ran in deferrable mode

Reply:
Changed here: https://github.com/apache/airflow/pull/31018/commits/110a8f976722b253eda5f9cd2ff9ef1a9ddb2746